pallets/pools/src/lib.rs

```diff
@@ -628,6 +684,42 @@ pub mod pallet {
 				None => None,
 			};
 
+			for (tranche, tranche_input) in tranches.tranches.iter().zip(&tranche_inputs) {
+				let token_name: BoundedVec<u8, T::MaxTokenNameLength> = tranche_input
+					.clone()
+					.metadata
+					.token_name
+					.try_into()
+					.map_err(|_| Error::<T>::BadMetadata)?;
+
+				let token_symbol: BoundedVec<u8, T::MaxTokenSymbolLength> = tranche_input
+					.clone()
+					.metadata
+					.token_symbol
+					.try_into()
+					.map_err(|_| Error::<T>::BadMetadata)?;
+
+				let decimals = match T::AssetRegistry::metadata(&currency) {
+					Some(metadata) => metadata.decimals,
+					None => return Err(Error::<T>::BadMetadata.into()),
+				};
+
+				let parachain_id = T::ParachainId::get();
+
+				let metadata = create_asset_metadata(
+					decimals,
+					tranche.currency,
+					parachain_id,
+					token_name.to_vec(),
+					token_symbol.to_vec(),
+				)?;
+
+				assert_ok!(T::AssetRegistry::register_asset(
+					Some(tranche.currency),
+					metadata
+				));
```

Comment:
It's the first time I am seeing an `assert_ok` used in an extrinsic. This will panic if it fails, right? That should not happen, we need to handle the failure here. Something like:

```suggestion
				T::AssetRegistry::register_asset(
					Some(tranche.currency),
					metadata
				).map_err(|_| Error::<T>::FailedToRegisterTrancheMetadata)?;
```